controllers/vmware/vspherecluster_reconciler.go

```diff
@@ -371,7 +372,32 @@ func (r *ClusterReconciler) VSphereMachineToCluster(ctx context.Context, o clien
 
 // Returns the failure domain information discovered on the cluster
 // hosting this controller.
-func (r *ClusterReconciler) getFailureDomains(ctx context.Context) (clusterv1.FailureDomains, error) {
+func (r *ClusterReconciler) getFailureDomains(ctx context.Context, namespace string) (clusterv1.FailureDomains, error) {
+	// Determine the source of failure domain based on feature gates NamespaceScopedZone.
+	// If NamespaceScopedZone is enabled, use Zone which is Namespace scoped,otherwise use
+	// Availability Zone which is Cluster scoped.
+	failureDomains := clusterv1.FailureDomains{}
```

Comment:
Nit
```suggestion
	failureDomains := clusterv1.FailureDomains{}
	// Determine the source of failure domain based on feature gates NamespaceScopedZone.
	// If NamespaceScopedZone is enabled, use Zone which is Namespace scoped,otherwise use
	// Availability Zone which is Cluster scoped.
```